### DISCUSSION TITLE
Behavior of the `--download-only` flag

### DISCUSSION THREAD

**Discussion by kvalev** [abraunegg/onedrive#1264] [Q&A] [answered]
Hi,

I have been running the client for a while now, usually with the `--download-only` flag, as I am only interested in having a backup/mirror of my OneDrive data. I was reorganizing some folder and moving things around and noticed that my local mirror did not remove the files, which were moved, so essentially I have two copies of a bunch of files.
I already deleted the sqlite database and re-triggered the sync process, but that didnt help and the files are still there. Is this how the flag is supposed to behave? Only download stuff, but never remove it locally?

**Comment by abraunegg** [comment, marked as answer]
@kvalev 
Correct - that flag will just download the files, not perform any local maintenance of local moves or deletes

**Reply by dtto** [reply]
Is there any way to do a local mirror where there are never local changes? That is, I want to pull the onedrive remote version to a local server but I want to ignore any local changes. I was also trying to use download-only in this way. I'd also recommend adding to the documentation that the download-only flag does not delete local files.

Thank you!

**Reply by abraunegg** [reply]
@dtto 
> Is there any way to do a local mirror where there are never local changes? That is, I want to pull the onedrive remote version to a local server but I want to ignore any local changes.

That would be the `--download-only` flag which would achieve this.

> I was also trying to use download-only in this way. I'd also recommend adding to the documentation that the download-only flag does not delete local files.

The `--download-only` replicates the remote state locally. If you are deleting local files post download, then this would be considered a local change, which you are ignoring by using this flag.

Feel free to submit a PR with documentation updates.

**Reply by dtto** [reply]
Apologies, I am not being clear. I thought you said that --download-only never deletes files even once removed from the remote? The local server that I am talking about has no local changes ever. It is solely a mirror. But the problem is, when files get deleted on onedrive, these deleted files remain on the local server. Is there a way to ensure that the local files are deleted when removed from the server while also ensuring that no local changes would get uploaded?

The reason for this is that while this machine is only a mirror, I want to avoid any issues where there is a filesystem problem on the local machine resulting in files being deleted on onedrive. That's why I'd like to just have it pull the remote state without pushing any local changes. Does that make sense? 

I looked at --no-remote-delete but this seems to be the opposite situation. I think what I want is to run with --no-local-delete but that flag doesn't exist.

**Reply by abraunegg** [reply]
@dtto 
> Apologies, I am not being clear. I thought you said that --download-only never deletes files even once removed from the remote? The local server that I am talking about has no local changes ever. It is solely a mirror. But the problem is, when files get deleted on onedrive, these deleted files remain on the local server. Is there a way to ensure that the local files are deleted when removed from the server 

So just use `--synchronize` so that your 'sync_dir' is kept in the correct state.

> while also ensuring that no local changes would get uploaded?

Hang on a second - you are saying that there are *no* changes locally ever ... now you are saying that there *are* local changes that you want to ignore .. ???

> I looked at --no-remote-delete but this seems to be the opposite situation. I think what I want is to run with --no-local-delete but that flag doesn't exist.

Hang on another second - you want a flag that is doing the exact of what you want - not deleting local files .. ??? 

The whole point of `--download-only` is to download your data for that point in time. If you want to ensure your local copy contains your deletes from OneDrive (moves|deletes performed online, or elsewhere and synced to OneDrive) as well - you need to use `--synchronize` to do this, then your local files will correctly replicate your online state. From my understanding this is what you want - and you dont want to keep an archive record of the files you have deleted online - which, currently your 'mirror' system is doing.

> The reason for this is that while this machine is only a mirror, I want to avoid any issues where there is a filesystem problem on the local machine resulting in files being deleted on onedrive.

Make sure your mirror system is built correctly:
* Use ZFS - mirrored pairs, raidz1,2,3 .. dont use anything else to store your data on, educate yourself as to why. Do not use anything else for data you care about.
* Configure your 'sync_dir' to use a separate ZFS volume, configure tools and ZFS options to snapshot your data
* Configure the 'onedrive' application to handle situations where your 'drives' or ZFS volume are unavailable - look at all the options regarding '.nosync' so that the application stops syncing if / when that flag is exposed. This can be used in the exact situation you are wanting to avoid - that is, file system issues poisoning your OneDrive environment - see: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-for-use-when-sync_dir-is-a-mounted-directory for details

**Reply by dtto** [reply]
Very helpful, thank you!
